src/dataconnect/types.ts

```diff
@@ -48,12 +48,16 @@ export interface File {
   content: string;
 }
 
-// An error indicating that the SQL database schema is incomptible with a data connect schema.
+// An error indicating that the SQL database schema is incompatible with a data connect schema.
 export interface IncompatibleSqlSchemaError {
-  // A list of differences between the two schema with instrucitons how to resolve them.
+  // A list of differences between the two schema with instructions how to resolve them.
   diffs: Diff[];
   // Whether any of the changes included are destructive.
   destructive: boolean;
+
+  // The failed precondition validation type.
+  // Either "INCOMPATIBLE_SCHEMA" or "INACCESSIBLE_SCHEMA".
+  violationType: string;
```

Comment:
```suggestion
  violationType: "INCOMPATIBLE_SCHEMA" | "INACCESSIBLE_SCHEMA" | string;
```

Good spot for a string union type, tho we likely should keep the 'string' in there since there's no guarantee other types wouldn't be returned in the future.